CHANGELOG.md

```diff
@@ -0,0 +1,135 @@
+  ([#2560](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2560))
+- Fix context leak in AWS SDK 2.2 and RocketMQ instrumentations
+  ([#2637](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2637))
+- Fix hang when a webflux http request is made inside of another webflux http request
+  (e.g. auth filter)
+  ([#2646](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2646))
+- Fix `@WithSpan` instrumentation breaking Java 6 classes
+  ([#2699](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2699))
+- Fix context not propagated over JMS when explicit destination used
+  ([#2702](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2702))
+- Fix StackOverflowError if jdbc driver implementation of Connection getMetaData calls
+  Statement execute
+  ([#2756](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2756))
+
+### 🧰 Tooling
+
+- Run smoke tests on Windows too!
+  ([#2568](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2568),
+   [#2621](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2621),
+   [#2617](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2617))
+- Added concurrent http client test with connection reuse
+  ([#2550](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2550),
+   [#2630](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2630),
+   [#2647](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2647))
+- Verify in smoke tests that runtime metrics are exported
+  ([#2603](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2603))
+- Remove JCenter which is about to be sunset
+  ([#2387](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2387))
+- Make sure muzzle build-time check actually validates anything
+  ([#2599](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2599))
+- Use Caffeine for weak maps
+  ([#2601](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2601))
+- Reimplement finding open ports
+  ([#2629](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2629))
+- Make muzzle reference creation package(s) configurable
+  ([#2615](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2615))
+- New Instrumenter API
+  ([#2596](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2596))
+- Added Concurrent server test
+  ([#2680](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2680),
+   [#2750](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2750),
+   [#2749](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2749))
+- Added smoke tests for Tomcat 10 and Jetty 11 (Servlet API 5)
```

Comment:
I wouldn't put this kind of thing in the changelog. end-users won't care about this, I don't think. I personally try to keep the changelog user-relevant, rather than inundating them with things they probably don't care about.

Reply:
I kinda agree with John here. Probably only "New Instrumenter API" is worth mentioning here.

Reply:
@anuraaga I think you introduced the "Tooling" category in the release notes, so want to check what you think.

I thought the "Tooling" category was nice for contributors who don't follow every PR, but it's certainly less work not to include it, and I agree it isn't relevant to end users.

Reply:
I think it's ok to have it since if it's at the end, easy enough to skip. The main motivation for having it was less about explaining to users and more about giving credit to the contributors, which I think is important, especially if they're non-maintainers. It could be only in release notes and not in the CHANGELOG, but I guess it's simpler if we just keep them matched up.

Reply:
I do agree with giving credit to contributors, but I think we need more clear understanding, who is the target group of release notes. Or having different sections for different groups.